README.md

```diff
@@ -1,20 +1,60 @@
-# Tripper
+Tripper
+=======
+_Triplestore wrapper for Python providing a simple and consistent interface to a range of triplestore backends._
 
-**A triplestore wrapper for Python.**
+[![PyPI](https://img.shields.io/pypi/v/tripper?logo=pypi)](https://pypi.org/project/tripper)
+[![Documentation](https://img.shields.io/badge/documentation-informational?logo=github)](https://emmc-asbl.github.io/tripper/latest/)
+![CI tests](https://github.com/EMMC-ASBL/tripper/workflows/CI%20-%20Tests/badge.svg)
 
-_The best ride when handling any triplestore._
 
-See the [documentation](docs/overview.md) for a quick tutorial.
+Basic concepts
+--------------
+**Tripper** strives for simplicity and is modelled after [rdflib] (with a few simplifications).
 
+In Tripper are:
+* All IRIs are represented by Python strings.
+  Example: `"http://emmo.info/emmo#Atom"`
 
-## Installation
+* Blank nodes are strings starting with "_:".
+  Example: `"_:bnode1"`
 
-The package can be installed from [PyPI](https://pypi.org/project/tripper) using `pip`:
+* Literals are constructed with [`tripper.Literal`][Literal].
+  Example: `tripper.Literal(3.14, datatype=XSD.float)`
+
+To make it easy to work with IRIs, provide Tripper a set of pre-defined namespaces, like `XSD.float`.
+New namespaces can easily be defined with the [`tripper.Namespace`][Namespace] class.
+
+A triplestore wrapper is created with the [`tripper.Triplestore`][Triplestore] class.
+
+
+
+Documentation
+-------------
+* Getting started: Take a look at the [tutorial](docs/tutorial.md).
+* Reference manual: [API Reference]
+
+
+Installation
+------------
+**Tripper** has by itself no dependencies outside the standard
+library, but the triplestore backends may have.
+
+
+The package can be installed from [PyPI] using `pip`:
 
 ```shell
 pip install tripper
 ```
 
-## License and copyright
+License and copyright
+---------------------
+All files in this repository are licensed under the [MIT license](LICENSE) with copyright &copy; 2022 SINTEF.
+
+
```

Comment:
I think this is a too strong claim. If other parties are contributing backends, they should keep the ownership on their contributions. But SINTEF as the copyright holder of the overall Tripper package can require that all contributed backends must have a MIT license.

Reply:
I agree - _however_ aren't we going to implement a plugin-system similar to the one for `oteapi-core` for adding backends? Or are we instead going to implement a "create a PR for `EMMC-ASBL/tripper` for adding and contributing a specific backend" approach? If the latter is indeed the case, we can look to ASE and how they add copyright and license within each of the contributed files.

Reply:
Also, you need to merge in the latest changes in `main` to get the updated copyright information that exists there.

Reply:
> Also, you need to merge in the latest changes in `main` to get the updated copyright information that exists there.

done

Reply:
> I agree - _however_ aren't we going to implement a plugin-system similar to the one for `oteapi-core` for adding backends? Or are we instead going to implement a "create a PR for `EMMC-ASBL/tripper` for adding and contributing a specific backend" approach? If the latter is indeed the case, we can look to ASE and how they add copyright and license within each of the contributed files.

Lets not spend time now implementing a complicated plugin system. With the ontorec and sparqlwrapper plugins, I think we cover most needs for plugins. 

I updated the copyright notice to make copyright (c) SINTEF the default if not otherwise stated in the top of the file.

Reply:
> Lets not spend time now implementing a complicated plugin system. (...)

As one would be a copy-paste of what has already been developed and implemented in OTEAPI Core, I don't think it will be time-consuming, nor complicated. Unless you, of course, think the OTEAPI Core plugin implementation is complicated...

> With the ontorec and sparqlwrapper plugins, I think we cover most needs for plugins.

If this is truly true, then that's great - although it doesn't really change the fact that a proper set of guidelines and rules should be setup for how to add more backends, how they should be implemented into the core code, and what functionality they should all, at minimum, share and implement.

Reply:
I agree that we need better guidelines (although the interface.py file already provides some formal guidelines). But I don't think that they need to go into the README file.